### PR TITLE
EMS109-Hr can see all employee daily status

### PR DESCRIPTION
**Pull request by rorpyll**

### CODE REVIEW

File: app/controllers/daily_tasks_controller.rb [hunk 1]
```diff
@@ -1,49 +1,73 @@
...
+
+  def show
+    @daily_task = if current_employee.is_hr?
+                    DailyTask.find_by_id params[:id]
+                  else
+                    @daily_task = current_employee.daily_tasks.find_by_id params[:id]
+
+                    redirect_to root_path, { notice: 'You Have Not Access :)' } unless @daily_task.present?
+
+                    @daily_task
```
Comment (salvimohan) [inline app/controllers/daily_tasks_controller.rb:22]: improve it

File: app/controllers/daily_tasks_controller.rb [hunk 2]
```diff
@@ -1,49 +1,73 @@
...
+    @daily_task = current_employee.daily_tasks.where('EXTRACT(DAY FROM created_at) = ?', Date.current.day).first
+    if @daily_task.present?
+      render :edit
+    else
+      redirect_to daily_tasks_path
+    end
+  end
+
+  def update
+    @daily_task = DailyTask.find(params[:id])
```
Comment (salvimohan) [inline app/controllers/daily_tasks_controller.rb:54]: daily status should be updated for employee of current date

File: app/views/daily_tasks/index.html.erb [hunk 3]
```diff
@@ -32,7 +32,8 @@
                   <tr>
                       <th scope="row"></th>
                      	<td><%= task.created_at.to_date%></td>
-                      <td ><button type="button" class="btn btn-info btn-sm" data-toggle="modal" data-target="#myModal-<%=task.id%>"><%= raw task.description.slice(0,15) + "..." %></button></td>  
+                      <td ><button type="button" class="btn btn-info btn-sm" data-toggle="modal" data-target="#myModal-<%=task.id%>"><%= raw task.description.slice(0,15) + "..." %></button></td> 
+                      <td><% if current_employee.is_hr? %><%= task.employee.name %><% end %></td>
```
Comment (salvimohan) [inline app/views/daily_tasks/index.html.erb:36]: improve it

File: spec/controllers/daily_task_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe DailyTasksController, type: :controller do
+  let!(:designation) { FactoryBot.create(:designation) }
+  let!(:role) { FactoryBot.create(:role) }
```
Comment (salvimohan) [inline spec/controllers/daily_task_controller_spec.rb:7]: move role in Fixtures

File: spec/models/employee_spec.rb [hunk 5]
```diff
@@ -21,4 +21,8 @@
     employee.joining_date = nil
     expect(employee.save).to eq(false)
   end
+
+  it 'check role is hr' do
```
Comment (salvimohan) [inline spec/models/employee_spec.rb:25]: add both scenario if employee is hr or not

File: app/models/daily_task.rb [hunk 6]
```diff
@@ -1,4 +1,7 @@
+# frozen_string_literal: true
+
 class DailyTask < ApplicationRecord
-	belongs_to :employee
-	validates :description,  presence: true
+  belongs_to :employee
+  validates :description, presence: true
```
Comment (salvimohan) [inline app/models/daily_task.rb:5]: unit test case for it

File: app/views/daily_tasks/index.html.erb [hunk 7]
```diff
@@ -32,7 +32,11 @@
                   <tr>
                       <th scope="row"></th>
                      	<td><%= task.created_at.to_date%></td>
-                      <td ><button type="button" class="btn btn-info btn-sm" data-toggle="modal" data-target="#myModal-<%=task.id%>"><%= raw task.description.slice(0,15) + "..." %></button></td>  
+                      <td ><button type="button" class="btn btn-info btn-sm" data-toggle="modal" data-target="#myModal-<%=task.id%>"><%= raw task.description.slice(0,15) + "..." %></button></td> 
+                      <td><% if current_employee.is_hr? %>
+                            <%= task.employee.name %>
+                          <% end %>
```
Comment (salvimohan) [inline app/views/daily_tasks/index.html.erb:38]: Use partial for it

File: app/controllers/daily_tasks_controller.rb (hunk 2 above)
Comment (salvimohan) [reply app/controllers/daily_tasks_controller.rb:52]: it should be resolved

File: app/controllers/daily_tasks_controller.rb [hunk 8]
```diff
@@ -1,49 +1,70 @@
...
-        render :edit
+      @daily_task = current_employee.daily_tasks.find_by_id params[:id]
+
+      redirect_to root_path, { notice: 'You Have Not Access :)' } unless @daily_task.present?
     end
   end
 
-	private
+  def new
+    task = current_employee.daily_tasks.map { |t| t if t.created_at.to_date == Time.now.to_date }.compact
```
Comment (salvimohan) [inline app/controllers/daily_tasks_controller.rb:25]: why is not using where here

**Review by salvimohan — APPROVED** [review]

**Review by rorshaan — APPROVED** [review]
LGTM

**Pull request merged**